docs/reference/filters.md

```diff
@@ -19,15 +19,31 @@ all: * -> filter1 -> filter2 -> "http://127.0.0.1:1234/";
 ## Template placeholders
 
 Several filters support template placeholders (`${var}`) in string parameters.
-Template placeholder is replaced by the value that is looked up in filter context path parameters.
-Missing template placeholder value interpretation depends on the filter.
+
+Template placeholder is replaced by the value that is looked up in the following sources:
+
+* request headers (if starts with `request.header.` prefix, e.g `${request.header.Content-Type}` is replaced by `Content-Type` request header value)
+* request url query (if starts with `request.url_query.` prefix, e.g `${request.url_query.q}` is replaced by `q` query parameter value)
+* request cookies (if starts with `request.cookie.` prefix, e.g `${request.cookie.PHPSESSID}` is replaced by `PHPSESSID` request cookie value)
+* response headers (if starts with `response.header.` prefix, e.g `${response.header.Location}` is replaced by `Location` response header value)
+* filter context path parameters (e.g. `${id}` is replaced by `id` path parameter value)
+
+Missing value interpretation depends on the filter.
 
 Example route that rewrites path using template placeholder:
 
 ```
 u1: Path("/user/:id") -> setPath("/v2/user/${id}") -> <loopback>;
 ```
 
+Example route that rewrites query parameter into header:
+```
+q2h: Path("/oauth2/tokeninfo") && QueryParam("access_token") ->
```

Comment:
maybe we could use a more "innocent" example? Meaning where there is no access token in the URL, so that we don't encourage this practice.

Reply:
This does not encourage but rather acknowledges that such [not recommended practice](https://tools.ietf.org/html/rfc6750#section-2.3) may exist. 

I agree it makes documentation reader to raise eyebrows which is not good so I will change it to something neutral.